src/contract/mod.rs

```diff
@@ -279,7 +279,16 @@ impl<T: Transport> Contract<T> {
     }
 
     /// Find events matching the topics.
-    pub async fn events<A, B, C, R>(&self, event: &str, topic0: A, topic1: B, topic2: C) -> Result<Vec<R>>
+    pub async fn events<A, B, C, R>(
+        &self,
+        event: &str,
+        from_block: Option<BlockNumber>,
+        to_block: Option<BlockNumber>,
+        block_hash: Option<H256>,
```

Comment:
Having 7 parameters, and 3 of them being optional is a bit too much. I'd like to propose a helper struct with a builder to amalgamate them.
```rust
pub struct BlockRange {
  from_block: Option<BlockNumber>,
  to_block: Option<BlockNumber>,
  block_hash: Option<H256>,
}

impl BlockRange {
  // set both `from_block` and `to_block`
  pub fn range(from: BlockNumber, to: BlockNumber) -> Self;
  // set just `from_block`
  pub fn from(BlockNumber) -> Self;
  // set just `to_block`
  pub fn to(BlockNumber) -> Self;
  // set just `block_hash`
  pub fn hash(H256) -> Self;
  // look for one exact block (set both `from/to` and `hash`).
  pub fn exact(BlockNumber, H256) -> Self;

  // applies the parameters to given filter
  pub (crate) fn apply_to(&self, b: FilterBuilder) -> FilterBuilder;
}
```